### ISSUE TITLE
pass calculator options through `pyhf.infer.upperlimit` (toys)

### ISSUE THREAD

**Issue by lukasheinrich** [scikit-hep/pyhf#1336]
# Description

Currently there's no easy way to pass custom options to upperlimit so it'll always acll asymptotics

**Comment by matthewfeickert** [comment]
@lukasheinrich I think this is a duplicate of Issue #1326, so I'm going to close in favor of that Issue. If you meant something more general feel free to reopen.

**Comment by matthewfeickert** [comment]
This actually hits on toys vs. asymptotics, which is slightly different. A PR could hit both this and Issue #1326.

**Comment by matthewfeickert** [comment]
Resolved in PR #1613 by @aryan26roy.

**Comment by matthewfeickert** [comment]
@lukasheinrich PR #1613 enabled this so we can now do

```python
import numpy as np
import pyhf

pyhf.set_backend("jax")
model = pyhf.simplemodels.uncorrelated_background(
    signal=[12.0, 11.0], bkg=[50.0, 52.0], bkg_uncertainty=[3.0, 7.0]
)
observations = [51, 48]
data = pyhf.tensorlib.astensor(observations + model.config.auxdata)
scan = np.linspace(0, 5, 11)
obs_limit, exp_limits, (scan, results) = pyhf.infer.intervals.upperlimit(
    data, model, scan, return_results=True, calctype="toybased", ntoys=2500
)
print(f"Observed limit: {obs_limit:.7f}")  # Observed limit: 0.9800679
for n_sigma, exp_limit in zip([-2, -1, 0, 1, 2], exp_limits):
    print(f"Expected {n_sigma} sigma limit: {exp_limit:.7f}")
# Expected -2 sigma limit: 0.4986645
# Expected -1 sigma limit: 0.8672213
# Expected 0 sigma limit: 0.9975027
# Expected 1 sigma limit: 1.5754683
# Expected 2 sigma limit: 1.9891135
```

I'll make sure to point this out in the release notes.
